cider/exec/nextgen/context/RuntimeContext.h

```diff
@@ -37,6 +37,14 @@ class RuntimeContext {
 
   void instantiate(const CiderAllocatorPtr& allocator);
 
+  // TBD: Currently, last batch would be output batch under all known scenarios.
+  Batch* getOutputBatch() {
```

Comment:
I am confusing about what's `RuntimeContext` for. seems `getOutputBatch` is a user side API, but when will `addBatch` `instantiate` be called? if these API are not used outside, better to use friend class for library internal call?

Reply:
```RuntimeContext``` played as a runtime objects holder, such as ```ArrowArray```, ```Buffer```, ```Hashtable```... You're right, we could hide these internal interfaces by friend class.